calculator/calculator.py

```diff
@@ -0,0 +1,218 @@
+"""Calculator module.
+
+This module allows evaluation of strings which represent mathematical
+expressions. The supported operations are declared in operators dictionary.
+
+Example:
+    To use this module, import the evaluate() function:
+
+        from calculator import evaluate
+
+    The parameter of this function is a mathematical expression (str). For
+    example, "1 + (3^-2) * (4! - 5$ 2)".
+"""
+
+import re
+import math
+import operator
+from sys import maxsize
+
+any_number_regex = r"-?\d+(\.\d+)?"
+
+
+class Operator:
+    """Representation of a single mathematical operator.
+
+    Attributes:
+        operation (function): the function that does the operation.
+        precedence (int): the priority of the operation.
+        regex (str): regex to find a pattern which suits to the operator.
+
+    Note:
+        Precedence determines which operation goes after another.
+    """
+
+    def __init__(self, operation, precedence, regex):
+        self.operation = operation
+        self.precedence = precedence
+        self.regex = re.compile(regex)
+
+
+def average(num1, num2):
+    """Return the average of two numbers (int/float).
+
+    Note:
+        Same as using mean() from statistics module.
```

Comment:
then use it

Reply:
Can't just implement a simple lambda?

Reply:
> then use it

Since mean accepts only lists, i rather that he will implement his function, this helps in the future.